support-frontend/assets/pages/new-contributions-landing/contributionsLandingInit.js

```diff
@@ -63,13 +64,25 @@ function getInitialPaymentMethod(
   );
 }
 
-function getInitialContributionType(countryGroupId: CountryGroupId): ContributionType {
-  const contributionType = getContributionTypeFromUrlOrElse(getContributionTypeFromSessionOrElse('MONTHLY'));
+function getInitialContributionType(
+  countryGroupId: CountryGroupId,
+  contributionTypes: ContributionTypes,
+): ContributionType {
+  const getDefaultContributionType = () => {
+    const defaultContributionType = contributionTypes[countryGroupId].find(ct => ct.isDefault);
+    return defaultContributionType ?
+      defaultContributionType.contributionType :
+      contributionTypes[countryGroupId][0].contributionType;
+  };
+
+  const contributionType =
+    getContributionTypeFromUrlOrElse(getContributionTypeFromSessionOrElse(getDefaultContributionType()));
+
   return (
     // make sure we don't select a contribution type which isn't on the page
-    getValidContributionTypes(countryGroupId).includes(contributionType)
+    contributionTypes[countryGroupId].find(ct => ct.contributionType === contributionType)
```

Comment:
the overall logic of this function is a bit hard to follow for me now. especially the appearance of the `contributionTypes[countryGroupId][0].contributionType` fallback in two places - is there a way to clarify?

Reply:
I've just pushed https://github.com/guardian/support-frontend/pull/1758/commits/3c3495042def3dfdc794905cc47b12427e4d7f5b to attempt to address this.
It checks the url/session (which may or may not be valid) before using the settings.